### PR TITLE
Add CUDA Healpix pad routines

### PR DESCRIPTION
**Pull request by daviddpruitt**
This adds Cuda padding routines for healpix xy grids so they don't have to be installed as a separate package.

### CODE REVIEW

File: earth2grid/healpixpad.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
[the remark this answers is not in the document]
Comment (daviddpruitt) [reply earth2grid/healpixpad.py]: Everything's been moved into healpix, all the padding is now handled by healpix.pad

File: earth2grid/healpixpad.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    @staticmethod
+    def backward(ctx, grad):
+        pad = ctx.pad
+        grad = grad.contiguous()
+        out = healpixpad_cuda.backward(grad, pad)[0]
+        return out, None
+
+
+class HEALPixPad(torch.nn.Module):
```
[the remark this answers is not in the document]
Comment (daviddpruitt) [reply earth2grid/healpixpad.py]: Replaced with a function, anyone using this as an object will need to update code but that's not a huge change.

File: setup.py [hunk 3]
```diff
@@ -63,7 +67,12 @@ def get_compiler():
             src_files,
             extra_compile_args=extra_compile_args,
             include_dirs=[os.path.abspath("earth2grid/csrc"), os.path.abspath("earth2grid/third_party/healpix_bare")],
-        )
+        ),
+        cpp_extension.CUDAExtension(
```
[the remark this answers is not in the document]
Comment (daviddpruitt) [reply setup.py]: Done

File: earth2grid/healpixpad.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nbren12) [reply earth2grid/healpixpad.py]: Thanks! Hopefully this is not too hard to implement downstream.

**Comment by nbren12** [comment]
Thanks @daviddpruitt! I fixed the issue with the CI. The build was failing since CUDA is not installed in GitHub actions.

**Comment by nbren12** [comment]
I think the matplotlib issue is fixed in #10, so will go ahead and merge. The test pass on my laptop.
